packages/ckeditor5-engine/src/controller/datacontroller.js

```diff
@@ -151,10 +151,11 @@ export default class DataController {
 	 * @param {String} [options.trim='empty'] Whether returned data should be trimmed. This option is set to `empty` by default,
 	 * which means whenever editor content is considered empty, an empty string will be returned. To turn off trimming completely
 	 * use `'none'`. In such cases exact content will be returned (for example `<p>&nbsp;</p>` for an empty editor).
+	 * @param {Object} [options.conversionOptions] Additional, custom configuration passed to the conversion process.
```

Comment:
Okay, let's talk about this. We can have one of three:

1.  The way you implemented it (and we think about the name).
2.  Having just one flat object (`{ root: 'main', showSuggestions: true, foo: 'bar' }`).
3.  Having two parameters (and we think about names).

What I don't like about current solution is that you need to specify additional "level" of object if you want to pass "conversion option":

`editor.getData( { conversionOptions: { showSuggestions: true } } )`

I don't like it. Let's talk about other options.

Having two parameters isn't perfect either:

`editor.getData( {}, { showSuggestions: true } )`

Having it flat is a bit messy and may be difficult to document, but would be the easiest to use:

`editor.getData( { showSuggestions: true } )`  
`editor.getData( { root: 'other' } )`

Reply:
And when it comes to name, it should make sense from the user's perspective. `conversionOptions` doesn't make a lot of sense, especially if someone is not a programmer, does not know CKE5 internals and simply integrates editor and wants to use `.getData()`.

`resultOptions`, `resultConfiguration`, `dataOptions`, `dataConfiguration` is the direction I'd go.

Or just having it all in a one, flat options object :shrug:

Reply:
I think that having one object would be best. For example, an option like `options.trim` also make sense to be available during conversion. From the integrator's perspective, one will not be able to understand why `showSuggestions` is a conversion option and `trim` is not. This is an implementation detail that should be abstracted.

That's why one `options` object should be enough.